### PR TITLE
Refact getJoinKindAndBuildSideIndex function

### LINKED ISSUE #7320: Refact function getJoinKindAndBuildSideIndex
## Enhancement
Current implementation can be improved in two  points:
1. Use only three maps to get mapping directly, instead of doing specific handlings after map getting ops.
2. Throw exceptions when invoked with unsupported <join kind, inner index>

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: close #7320

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Coprocessor/JoinInterpreterHelper.cpp [hunk 1]
```diff
@@ -38,29 +38,54 @@ namespace JoinInterpreterHelper
...
+        {{tipb::JoinType::TypeRightOuterJoin, 0}, {ASTTableJoin::Kind::Cross_LeftOuter, 0}},
+        {{tipb::JoinType::TypeRightOuterJoin, 1}, {ASTTableJoin::Kind::Cross_LeftOuter, 0}},
+        {{tipb::JoinType::TypeSemiJoin, 1}, {ASTTableJoin::Kind::Cross, 1}},
+        {{tipb::JoinType::TypeAntiSemiJoin, 1}, {ASTTableJoin::Kind::Cross_Anti, 1}},
+        {{tipb::JoinType::TypeLeftOuterSemiJoin, 1}, {ASTTableJoin::Kind::Cross_LeftOuterSemi, 1}},
+        {{tipb::JoinType::TypeAntiLeftOuterSemiJoin, 1}, {ASTTableJoin::Kind::Cross_LeftOuterAnti, 1}}};
+    static const std::unordered_map<std::pair<tipb::JoinType, size_t>, std::pair<ASTTableJoin::Kind, size_t>, JoinKindAndInnerIndexPairHash> null_aware_join_type_map{
+        {{tipb::JoinType::TypeAntiSemiJoin, 1}, {ASTTableJoin::Kind::NullAware_Anti, 1}},
+        {{tipb::JoinType::TypeLeftOuterSemiJoin, 1}, {ASTTableJoin::Kind::NullAware_LeftOuterSemi, 1}},
+        {{tipb::JoinType::TypeAntiLeftOuterSemiJoin, 1}, {ASTTableJoin::Kind::NullAware_LeftOuterAnti, 1}}};
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/JoinInterpreterHelper.cpp:86]: Does that mean that for `null_aware_join`, the build side index must be 1?

Comment (yibin87) [reply dbms/src/Flash/Coprocessor/JoinInterpreterHelper.cpp:86]: Yes, tidb ensures this for us: if isSemiJoin() fixBuildSide = true :)

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**